streams/shared/src/main/scala/zio/stream/experimental/ZSink.scala

```diff
@@ -342,6 +342,14 @@ object ZSink {
     new ZSink(loop(Chunk.empty))
   }
 
+  /**
+   * A sink that collects all of its inputs into chunks of maximum size `n`.
```

Comment:
Can you also add a note that every chunk is preallocated and must fit in memory? Or @adamgfraser should we use the same approach you arrived at with regards to the size hint? (Which I don't remember what it was 😅)

Reply:
I think I've come to the point of view that we should just hint the specified size if an operator allows specifying a particular chunk size. I think we have to do that for maximum efficiency and it is just user error if you specify a chunk size like `Int.MaxValue`.

I was a little hesitant on this because of an issue we had with someone blowing themselves up using `queue.takeUpTo(Int.MaxValue)` but I think that is different in that there the `n` is how many elements to take and chunking is an implementation detail, whereas here the whole point of the parameter is to specify the size of the chunk to take.